src/lib.rs

```diff
@@ -20,8 +20,8 @@ impl Infinispan {
         base_url: impl Into<String>,
         username: impl AsRef<str>,
         password: impl AsRef<str>,
-    ) -> Infinispan {
-        Infinispan {
+    ) -> Self {
```

Comment:
I don't have a strong preference about this. Why do you prefer `Self` is it just because it makes renaming easier?

Reply:
Yes. But not only because of that, but because it's the adopted idiomatic convention.